### PR TITLE
ALS-7014: Implement signed URL functionality for data exports

### PR DESCRIPTION
**Pull request by ramari16**

### CODE REVIEW

File: service/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/service/PicSureService.java [hunk 1]
```diff
@@ -237,6 +240,37 @@ public ResponseEntity queryResult(@PathVariable("resourceQueryId") UUID queryId,
...
+		AsyncResult result = queryService.getResultFor(queryId.toString());
+		if (result == null) {
+			// This happens sometimes when users immediately request the status for a query
+			// before it can be initialized. We wait a bit and try again before throwing an
+			// error.
+			try {
+				Thread.sleep(100);
+			} catch (InterruptedException e) {
+				return ResponseEntity.status(500).build();
+			}
```
Comment (Luke-Sikina) [inline service/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/service/PicSureService.java:254]: What does waiting do? Are you debouncing a button on the backend?

File: service/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/service/PicSureService.java [hunk 2]
```diff
@@ -237,6 +240,37 @@ public ResponseEntity queryResult(@PathVariable("resourceQueryId") UUID queryId,
...
+		if (result.getStatus() == AsyncResult.Status.SUCCESS) {
+			File file = result.getFile();
+			signUrlService.uploadFile(file, file.getName());
+			String presignedGetUrl = signUrlService.createPresignedGetUrl(file.getName());
+			log.info("Presigned url: " + presignedGetUrl);
+			return ResponseEntity.ok()
+					.contentType(MediaType.APPLICATION_JSON)
+					.body(new SignedUrlResponse(presignedGetUrl));
+		} else {
+			return ResponseEntity.status(400).body("Status : " + result.getStatus().name());
```
Comment (Luke-Sikina) [inline service/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/service/PicSureService.java:270]: Is 400 a fair response code for pending statuses? Maybe 202?

**Review by Luke-Sikina — APPROVED** [review]

File: service/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/service/PicSureService.java (hunk 2 above)
Comment (ramari16) [reply service/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/service/PicSureService.java:270]: Yes, I think it is appropriate. Or at least consistent with the current workflow of the /result endpoint. The front end polls /status until it is SUCCESS, then calls this endpoint for a signed url, or /result for the document itself

File: service/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/service/PicSureService.java (hunk 1 above)
Comment (ramari16) [reply service/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/service/PicSureService.java:254]: This is boilerplate code copied from the /result endpoint. But true, it should not be needed, no one should call this endpoint until the status returns SUCCESS which would mean the query exists

**Pull request merged**